import_ark_bn254/src/lib.rs

```diff
@@ -0,0 +1,45 @@
+#![no_std]
+use ark_bn254::{Bn254, Fq12, G1Affine, G2Affine};
+use ark_ec::pairing::Pairing;
+use ark_ff::Field;
+use ark_serialize::CanonicalDeserialize;
+use soroban_sdk::{contract, contractimpl, contracttype, BytesN, Env};
+
+// Soroban contracts do not supports `std` features. To use dynamic memory
+// allocation (which is required for ark libraries), import the `alloc` crate
+// (the `GlobalAlloc` impl is provided by the `soroban_sdk`).
+//
+// For more context refer to the [Contract Rust
+// Dialect](https://developers.stellar.org/docs/learn/encyclopedia/contract-development/rust-dialect)
+// doc.
+extern crate alloc;
```

Comment:
I don't see us using the alloc namespace in this file. Is this line needed?

Reply:
Tried locally, and removing this line doesn't affect anything, and the builds and tests run fine. So I think we can remove it.